packages/backend-common/src/reading/GerritUrlReader.ts

```diff
@@ -14,23 +14,38 @@
  * limitations under the License.
  */
 
-import { NotFoundError } from '@backstage/errors';
+import { Git } from '../scm';
+import { NotFoundError, NotModifiedError } from '@backstage/errors';
 import {
   GerritIntegration,
+  getGerritCloneRepoUrl,
+  getGerritBranchApiUrl,
   getGerritFileContentsApiUrl,
   getGerritRequestOptions,
+  parseGerritJsonResponse,
+  parseGitilesUrl,
 } from '@backstage/integration';
+import fs from 'fs-extra';
 import fetch, { Response } from 'node-fetch';
+import os from 'os';
+import { join as joinPath } from 'path';
+import tar from 'tar';
+import { Readable } from 'stream';
 import {
   ReaderFactory,
+  ReadTreeOptions,
   ReadTreeResponse,
+  ReadTreeResponseFactory,
   ReadUrlOptions,
   ReadUrlResponse,
   SearchResponse,
   UrlReader,
 } from './types';
 import { ScmIntegrations } from '@backstage/integration';
 
+const createTemporaryDirectory = async (): Promise<string> =>
+  await fs.mkdtemp(joinPath(os.tmpdir(), '/gerrit-clone-'));
```

Comment:
We'll want to use the `backend.workingDirectory` in config if it's there. You can see how it's set up for example [here](https://github.com/backstage/backstage/blob/e1e06cf91361751b02f8c7f31443c84ba7ecb6df/packages/backend-common/src/reading/tree/ReadTreeResponseFactory.ts#L32-L34)